### PR TITLE
Premium content: Add paid flag

### LINKED ISSUE #41584: Premium Content: Paid flag in the block inserter
All the blocks that require upgrading have a paid flag in the block inserter. We need to add one to the Premium content block.

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request

Appends a "paid" flag to the Premium Content block title displayed on the block inserter when the site requires a upgrade.

Screen Shot 2020-05-28 at 16 03 44
Screen Shot 2020-05-28 at 16 04 52

#### Testing instructions

* Apply D44086-code or `cd apps/full-site-editing; yarn dev --sync`.
* Sandbox the API.
* Switch to a sandboxed site under a Free plan.
* Add a new post.
* Open the block inserter and search the Premium Content block.
* Make sure the title contains a "paid" flag at the end.
* Upgrade to a paid plan or switch to a sandboxed site under a paid plan.
* Add a new post.
* Open the block inserter and search the Premium Content block.
* Make sure the title does not contain a "paid" flag at the end.

Fixes #41584

### CODE REVIEW

File: apps/full-site-editing/full-site-editing-plugin/premium-content/index.js [hunk 1]
```diff
@@ -33,11 +39,40 @@ const registerBlock = ( block ) => {
...
+			return;
+		}
+
+		const paidFlag = _x(
+			'paid',
+			'Short label appearing near a block requiring a paid plan',
+			'premium-content'
+		);
+
+		unregisterBlockType( container.name );
```
Comment (mmtr) [inline apps/full-site-editing/full-site-editing-plugin/premium-content/index.js:60]: I wish there was some other way of updating the settings of a block already registered.

**Comment by kwight** [comment]
Hm, would this be a premature discouragement to the user? Should we instead want them to add the block, see what it's all about, and then consider the upgrade? I can see it both ways, depending on whether I have my user hat or my marketing hat on 🙃 (I guess that also brings up the question of adding it to the Recurring Payments block in JP?)

Works well though 👍

File: apps/full-site-editing/full-site-editing-plugin/premium-content/index.js [hunk 2]
```diff
@@ -33,11 +39,40 @@ const registerBlock = ( block ) => {
 	} );
 };
 
+/**
+ * Appends a "paid" tag to the Premium Content block title if site requires an upgrade.
+ */
+const addPaidBlockFlags = async () => {
+	const membershipsStatus = await apiFetch( { path: '/wpcom/v2/memberships/status' } );
+	const shouldUpgrade = membershipsStatus.should_upgrade_to_access_memberships;
```
Comment (gwwar) [inline apps/full-site-editing/full-site-editing-plugin/premium-content/index.js:47]: What happens if the fetch fails?

File: apps/full-site-editing/full-site-editing-plugin/premium-content/index.js [hunk 3]
```diff
@@ -33,11 +39,40 @@ const registerBlock = ( block ) => {
...
+	if ( shouldUpgrade ) {
+		const premiumContentBlock = getBlockType( container.name );
+		if ( ! premiumContentBlock ) {
+			return;
+		}
+
+		const paidFlag = _x(
```
Comment (gwwar) [inline apps/full-site-editing/full-site-editing-plugin/premium-content/index.js:54]: I had to look up this one. Signature is _x( $text, $context, $domain ). The advantage over `__` is translator context and multiple translations per context.

https://developer.wordpress.org/reference/functions/_x/

File: apps/full-site-editing/full-site-editing-plugin/premium-content/index.js (hunk 2 above)
Comment (gwwar) [reply apps/full-site-editing/full-site-editing-plugin/premium-content/index.js:47]: Oh interesting! It adds an inline error to the block insert:

Screen Shot 2020-05-28 at 11 14 52 AM

Screen Shot 2020-05-28 at 11 15 10 AM

File: apps/full-site-editing/full-site-editing-plugin/premium-content/index.js (hunk 3 above)
Comment (mmtr) [reply apps/full-site-editing/full-site-editing-plugin/premium-content/index.js:54]: TBH I copied from what we do for the Jetpack block:

https://github.com/Automattic/jetpack/blob/796147bfa0fd7d15661ca41337a37f05ca19cf69/extensions/shared/register-jetpack-block.js#L17

**Comment by mmtr** [comment]
> I guess that also brings up the question of adding it to the Recurring Payments block in JP?

We already do it! This PR is to actually align the Premium Content block with what we currently do for paid Jetpack blocks like Recurring Payments or Calendly:

Screen Shot 2020-05-29 at 10 48 19

File: apps/full-site-editing/full-site-editing-plugin/premium-content/index.js (hunk 2 above)
Comment (mmtr) [reply apps/full-site-editing/full-site-editing-plugin/premium-content/index.js:53]: > It adds an inline error to the block insert

I think that's actually caused by [this other request](https://github.com/Automattic/wp-calypso/blob/0f5b4042c7996f88f38376ac501e047bb8a42149/apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/container/edit.js#L215-L218).

> Let's maybe not display paid in the error case and make sure no JS errors are thrown.

Yeah, I like this, so I added some error handling in 008680c to make sure the flag is not added and no JS errors are shown:

Screen Shot 2020-05-29 at 11 18 50

**Comment by artpi** [comment]
Fantastic, thank you!

**Pull request merged**